sdk/samples/core/mosquitto_multiple_rpc_sample.c

```diff
@@ -0,0 +1,629 @@
+      {
+        if (recv_data->status != AZ_MQTT5_RPC_STATUS_OK)
+        {
+          printf(
+              LOG_APP_ERROR "Error response received. Status :%d. Message :%s\n",
+              recv_data->status,
+              az_span_ptr(recv_data->error_message));
+        }
+        else
+        {
+          if (!az_span_is_content_equal(content_type, recv_data->content_type))
+          {
+            printf(
+                LOG_APP_ERROR "Invalid content type. Expected: {%s} Actual: {%s}\n",
+                az_span_ptr(content_type),
+                az_span_ptr(recv_data->content_type));
+          }
+          else
+          {
+            // TODO: deserialize before passing to handle_response
+            handle_response(recv_data->response_payload);
+          }
+        }
+        remove_command(&client_pending_commands, recv_data->correlation_id);
+      }
+      else
+      {
+        printf(LOG_APP_ERROR "Request with ");
+        print_correlation_id(recv_data->correlation_id);
+        printf("not found\n");
+      }
+      break;
+    }
+
+    case AZ_MQTT5_EVENT_RPC_CLIENT_ERROR_RSP:
+    {
+      az_mqtt5_rpc_client_rsp_event_data* recv_data
+          = (az_mqtt5_rpc_client_rsp_event_data*)event.data;
+      printf(LOG_APP_ERROR "Broker/Client failure for command ");
+      print_correlation_id(recv_data->correlation_id);
+      printf(": %s Status: %d\n", az_span_ptr(recv_data->error_message), recv_data->status);
+      remove_command(&client_pending_commands, recv_data->correlation_id);
+      break;
+    }
+
+    case AZ_HFSM_EVENT_ERROR:
+      printf(LOG_APP_ERROR "Error Event\n");
+      // az_platform_critical_error();
+      break;
+
+    default:
+      // TODO:
+      break;
+  }
+
+  return AZ_OK;
+}
+
+int main(int argc, char* argv[])
+{
+  (void)argc;
+  (void)argv;
+
+  /* Required before calling other mosquitto functions */
+  if (mosquitto_lib_init() != MOSQ_ERR_SUCCESS)
+  {
+    printf(LOG_APP "Failed to initialize MosquittoLib\n");
+    return -1;
+  }
+
+  printf(LOG_APP "Using MosquittoLib version %d\n", mosquitto_lib_version(NULL, NULL, NULL));
+
+  az_log_set_message_callback(az_sdk_log_callback);
+  az_log_set_classification_filter_callback(az_sdk_log_filter_callback);
+
+  connection_context = az_context_create_with_expiration(
+      &az_context_application, az_context_get_expiration(&az_context_application));
+
+  az_mqtt5 mqtt5;
+  struct mosquitto* mosq = NULL;
+
+  az_mqtt5_options mqtt5_options = az_mqtt5_options_default();
+  mqtt5_options.disable_tls = true;
+
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_init(&mqtt5, &mosq, &mqtt5_options));
+
+  az_mqtt5_connection_options connection_options = az_mqtt5_connection_options_default();
+  connection_options.client_id_buffer = client_id;
+  connection_options.username_buffer = username;
+  connection_options.password_buffer = AZ_SPAN_EMPTY;
+  connection_options.hostname = hostname;
+  connection_options.port = 1883;
```

Comment:
#define

Reply:
moved this sample to have the same default setup as other samples